### ISSUE TITLE
Telegram Desktop Local Passcode

### ISSUE THREAD

**Issue by ZX-zedex** [MihaZupan/TelegramDesktopLocalStorage#4]
I need to find this passcode in Telegram Desktop, which is stored in the map[0-1] file.

![image](https://user-images.githubusercontent.com/54175250/63956974-f2499f00-caa0-11e9-8b36-f967e66124e3.png)

I don't have any idea about the length of the password, which characters it contains and the strength of it.
Is there any chance to find out it without bruteforce method or maybe replace it in the map[0-1] file with the simplest one (ex. 1234).

Thank you.

**Comment by MihaZupan** [comment]
If you want access to the files you will have to find out the passcode. JohnTheRipper has support for bruteforcing it.

**Comment by ZX-zedex** [comment]
Yes. I know. But I can't cause the Desktop version supports various symbols (not just a 4-digit number) and I am not sure that it contains only english characters.
Bruteforcing does work but it is very slow and I don't think that in my life it will end)

That's why I wanted to know if there is a way to get the hash of the password maybe one can change the script to replace it with something that you want to input.

Thank you.

**Comment by MihaZupan** [comment]
You need the actual passcode to further decrypt files. I'm afraid bruteforcing is the only way.

**Comment by ZX-zedex** [comment]
Hi.
Thank you for your reply.
Is there any capable settings that can afford to bruteforce the hash with cyrillic symbols (with JTP)?

Thank you

**Comment by MihaZupan** [comment]
I would not know as I haven't used John the ripper all that much. The community on their github seems helpful though so you should try asking there.
